### PR TITLE
Generic credential display capabilities

### PR DESCRIPTION
**Pull request by tolson4**
Requires changes in https://github.com/digitalbazaar/bedrock-karma/pull/23 for unit tests to pass.

### CODE REVIEW

File: test/package.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
   "scripts": {
     "debug": "node --preserve-symlinks test.js test --framework karma --log-level debug",
     "test": "node --preserve-symlinks test.js test --framework karma",
-    "coverage": "node --preserve-symlinks test.js test --framework karma --config ./node_modules/@bedrock/karma/lib/coverageConfig.js",
-    "coverage-ci": "node --preserve-symlinks test.js test --framework karma --config ./node_modules/@bedrock/karma/lib/coverageConfig.js"
+    "coverage": "cross-env NODE_ENV=test c8 npm run test",
+    "coverage-ci": "cross-env NODE_ENV=test c8 --reporter=lcovonly --reporter=text-summary --reporter=text npm run test"
```
Comment (dlongley) [inline test/package.json:10]: @davidlehn, can you look at the changes in this file to make sure it looks good to you?

**Review by dlongley — CHANGES_REQUESTED** [review]
Thanks! This needs a changelog entry. My understanding is that because this uses the new `credential-switch` component, an app using this module would need changes, so it's a breaking change.

File: components/Register.vue [hunk 2]
```diff
@@ -183,6 +166,13 @@ export default {
     },
     maximizeModal() {
       return this.$q.screen.lt.md;
+    },
+    tos() {
+      const {vueWallet: {terms = ''}} = config;
+      if(terms.length === 0) {
+        return ['These are the terms of service.'];
```
Comment (davidlehn) [inline components/Register.vue:173]: But "These are the terms of service." are not terms of service. ;-)  Maybe the default could be something that stands alone like "Please contact us for the terms of service."?

Comment (davidlehn) [reply components/Register.vue:173]: I also assume at some point the terms will be HTML with headers and lists and such, so this line splitting paragraph method may have to go away.

Comment (tolson4) [reply components/Register.vue]: Updated in a7b29e2efdbfc2f78b0063d8dbc9aa3c8a15d2b4

> I also assume at some point the terms will be HTML with headers and lists and such, so this line splitting paragraph method may have to go away.

HTML is a good option. I could also see having a pre-prepared terms of service in PDF form could be desirable as well.

File: test/package.json (hunk 1 above)
Comment (davidlehn) [inline test/package.json:10]: c8 is for node code, we have some test karma commands for coverage, though nothing happens in the tests here so they do nothing:
```suggestion
    "coverage": "node --preserve-symlinks test.js test --framework karma-coverage",
    "coverage-ci": "node --preserve-symlinks test.js test --framework karma-coverage-ci"
```

Comment (davidlehn) [reply test/package.json:10]: This needs to use our karma coverage support, c8 is for node.

File: test/package.json [hunk 3]
```diff
@@ -34,7 +34,23 @@
     "@vuelidate/validators": "^2.0.0-alpha.29",
     "@vueuse/core": "^8.5.0",
     "ajv": "^6.12.6",
+    "c8": "^7.11.3",
+    "cross-env": "^7.0.3",
```
Comment (davidlehn) [inline test/package.json:38]:
```suggestion
```

File: test/package.json [hunk 4]
```diff
@@ -34,7 +34,23 @@
...
+      "node_modules/@bedrock/vue-wallet/**"
+    ],
+    "exclude": [
+      "node_modules/@bedrock/vue-wallet/node_modules/**"
+    ],
+    "reporter": [
+      "lcov",
+      "text-summary",
+      "text"
+    ]
```
Comment (davidlehn) [inline test/package.json:54]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
   },
   "peerDependencies": {
     "@bedrock/quasar": "^8.0.1",
-    "@bedrock/vue-credential-card": "^5.0.0",
+    "@bedrock/vue-credential-card": "github:digitalbazaar/bedrock-vue-credential-card#generic-credential-component",
```
Comment (davidlehn) [inline package.json:17]: This needs to be a released version before merging.

Comment (tolson4) [reply package.json:17]: Agreed. This library has been renamed and moved to here https://github.com/digitalbazaar/bedrock-vue-vc/pull/1.

File: test/package.json (hunk 1 above)
Comment (tolson4) [reply test/package.json]: Fixed here 4c5a720aff7f923d11e7a4ebe480e0f73b6abc55

File: package.json (hunk 5 above)
Comment (tolson4) [reply package.json]: resolved in https://github.com/digitalbazaar/bedrock-vue-wallet/pull/13/commits/b290b8b90a2cbad40b09691fea46fc6f4f845ec8

**Pull request merged**